src/ajax/map_variants.php

```diff
@@ -198,6 +198,195 @@ function lovd_mapVariantToTranscripts (&$aVariant, $aTranscripts)
 
 
 
+
+// Performing lift-overs so variants are described as relative to as many GBs as possible.
+$aActiveGBs = $_DB->query('
+        SELECT id,
+            CONCAT("VariantOnGenome/DNA", if(column_suffix = "", "", "/"), column_suffix) as "DNA",
+            CONCAT("position_g_start", if(column_suffix = "", "", "_"), column_suffix) as "position_start",
+            CONCAT("position_g_end", if(column_suffix = "", "", "_"), column_suffix) as "position_end"
+        FROM ' . TABLE_GENOME_BUILDS
+)->fetchAllGroupAssoc();
+
+if (count($aActiveGBs) < 1) {
+    // Lift overs can only be performed if more than one GB is active.
+
+    foreach ($aActiveGBs as $sBuild => $aGBColumns) {
+        $aChr = $_DB->query(
+            'SELECT DISTINCT chromosome FROM ' . TABLE_VARIANTS .
+            ' WHERE ' . $aGBColumns['DNA'] . '= ""' . // DNA field of this GB is empty
+            '    AND mapping_flags & ' . MAPPING_ALLOW . // Mapping is allowed
+            '    AND NOT mapping_flags & ' . (MAPPING_NOT_RECOGNIZED | MAPPING_DONE) // Mapping is possible and necessary
+        )->fetchColumn();
+
+        if (!empty($aChr)) {
+            // There is more than one chromosome with missing variant description for this build.
+
+            // We pick a random chromosome to minimize the chances of multiple active users
+            //  activating lift overs on the same chromosome.
+            $sChr = array_rand($aChr);
+
+            $aVariantIDs = $_DB->query(
+                'SELECT id FROM ' . TABLE_VARIANTS .
+                ' WHERE chromosome = ' . $sChr . // Only taking our picked chromosome
+                '    AND ' . $aGBColumns['DNA'] . '= ""' . // DNA field of this GB is empty
+                '    AND mapping_flags & ' . MAPPING_ALLOW . // Mapping  is allowed
+                '    AND NOT mapping_flags & ' . (MAPPING_NOT_RECOGNIZED | MAPPING_DONE) . // Mapping is possible and necessary
+                ' LIMIT ' . $nMaxVariants
+            )->fetchColumn();
+
+            // We will loop through all variants that are missing DNA descriptions.
+            foreach ($aVariantIDs as $sVariantID) {
+
+                // This variant is now in progress, which we will indicate by mapping flags.
+                $_DB->query(
+                    'UPDATE ' . TABLE_VARIANTS .
+                    ' SET mapping_flags = mapping_flags | ' . MAPPING_IN_PROGRESS .
+                    ' WHERE id = ?', array($sVariantID)
+                );
+
+                list($bMappingTryAgain, $bMappingSuccessful) = array(false, false);
```

Comment:
Easier is:
```php
$bMappingTryAgain = $bMappingSuccessful = false;
```
It doesn't require building an array and then decomposing it again.

Reply:
Oh great, thank you!